lib/services/parser.go

```diff
@@ -53,6 +53,93 @@ var (
 	CertAuthorityTypeExpr = builder.Identifier(`system.catype()`)
 )
 
+// PredicateAllEndWith is a custom function to test if a string ends with a
+// particular suffix. If given a `[]string` as the first argument, all values
+// must have the given suffix (2nd argument).
+func PredicateAllEndWith(a interface{}, b interface{}) predicate.BoolPredicate {
+	return func() bool {
+		// bval is the suffix and must always be a plain string.
+		bval, ok := b.(string)
+		if !ok {
+			return false
+		}
+
+		switch aval := a.(type) {
+		case string:
+			return strings.HasSuffix(aval, bval)
+		case []string:
+			for _, val := range aval {
+				if !strings.HasSuffix(val, bval) {
+					return false
+				}
+			}
+			return true
+		default:
+			return false
+		}
+	}
+}
+
+// PredicateAllEqual is a custom function to test if all entries in a []string
+// are equal to a certain value. This is primarily useful for comparing string
+// fields that are only expected to contain a single, specific value.
+func PredicateAllEqual(a interface{}, b interface{}) predicate.BoolPredicate {
```

Comment:
I'm split on whether or not this makes more sense as a seperate predicate, or if we should just modify `equals` to support this. I think you've made the right call, just figured its worth discussing.

Reply:
I think that could make it hard (maybe impossible) to compare a plain `string` to a value that you want to contain a `string` but actually contains a `[]string`. I don't think this situation is ever likely to happen since we very rarely compare dynamic values, but it's not impossible.